stl/src/winapinls.cpp

```diff
@@ -1,661 +0,0 @@
-            bottom = middle + 1;
-        }
-    }
-
-    return -1;
-}
-
-// __crtDownlevelLocaleNameToLCID() - Maps the locale name to LCID on Windows XP.
-extern "C" LCID __cdecl __crtDownlevelLocaleNameToLCID(LPCWSTR localeName) {
-    int index;
-
-    if (localeName == nullptr) {
-        return 0;
-    }
-
-    index = GetTableIndexFromLocaleName(localeName);
-
-    if (index < 0 || index >= _countof(LcidToLocaleNameTable)) {
-        return 0;
-    }
-
-    return LcidToLocaleNameTable[index].lcid;
-}
-
-// __crtDownlevelLCIDToLocaleName() - Maps the LCID to locale name on Windows XP.
-extern "C" int __cdecl __crtDownlevelLCIDToLocaleName(LCID lcid, LPWSTR outLocaleName, int cchLocaleName) {
-    if (lcid == 0 || lcid == LOCALE_USER_DEFAULT || lcid == LOCALE_SYSTEM_DEFAULT) {
-        return 0;
-    }
-
-    if ((outLocaleName == nullptr && cchLocaleName > 0) || cchLocaleName < 0) {
-        return 0;
-    }
-
-    const int index = GetTableIndexFromLcid(lcid);
-    if (index < 0) {
-        return 0;
-    }
-
-    const wchar_t* const buffer = LcidToLocaleNameTable[index].localeName;
-    // the following cast is safe because LOCALE_NAME_MAX_LENGTH fits in int:
-    const int count = static_cast<int>(wcsnlen(buffer, LOCALE_NAME_MAX_LENGTH));
-
-    if (cchLocaleName > 0) {
-        if (count >= cchLocaleName) {
-            return 0;
-        }
-
-        wcscpy_s(outLocaleName, cchLocaleName, buffer);
-    }
-
-    return count + 1;
-}
-
-// __crtCompareStringEx() - Wrapper for CompareStringEx().
-extern "C" int __cdecl __crtCompareStringEx(
-    LPCWSTR lpLocaleName, DWORD dwCmpFlags, LPCWSTR lpString1, int cchCount1, LPCWSTR lpString2, int cchCount2) {
-    // use CompareStringEx if it is available (only on Windows Vista+)...
-    IFDYNAMICGETCACHEDFUNCTION(PFNCOMPARESTRINGEX, CompareStringEx, pfCompareStringEx) {
-        return pfCompareStringEx(
-            lpLocaleName, dwCmpFlags, lpString1, cchCount1, lpString2, cchCount2, nullptr, nullptr, 0);
-    }
-
-    // ...otherwise fall back to using CompareString.
-    return CompareStringW(
-        __crtDownlevelLocaleNameToLCID(lpLocaleName), dwCmpFlags, lpString1, cchCount1, lpString2, cchCount2);
-}
-
-// __crtLCMapStringEx() - Wrapper for LCMapStringEx().
-extern "C" int __cdecl __crtLCMapStringEx(
-    LPCWSTR lpLocaleName, DWORD dwMapFlags, LPCWSTR lpSrcStr, int cchSrc, LPWSTR lpDestStr, int cchDest) {
-    // use LCMapStringEx if it is available (only on Windows Vista+)...
-    IFDYNAMICGETCACHEDFUNCTION(PFNLCMAPSTRINGEX, LCMapStringEx, pfLCMapStringEx) {
-        return pfLCMapStringEx(lpLocaleName, dwMapFlags, lpSrcStr, cchSrc, lpDestStr, cchDest, nullptr, nullptr, 0);
-    }
-
-    // ...otherwise fall back to using LCMapString.
-    return LCMapStringW(__crtDownlevelLocaleNameToLCID(lpLocaleName), dwMapFlags, lpSrcStr, cchSrc, lpDestStr, cchDest);
-}
-
-
-// __crtGetLocaleInfoEx() - Wrapper for GetLocaleInfoEx().
-extern "C" int __cdecl __crtGetLocaleInfoEx(

```

Comment:
It appears that this is changing the export surface, which is a bincompat break:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29115\bin\Hostx86\x86>dumpbin /exports msvcp140.dll | findstr __crtGetLocaleInfoEx
       1501  5DC 0000C1E0 __crtGetLocaleInfoEx
```

This function was declared as `_CRTIMP2`, which exports it. We need to be sure that the export surface of the DLL remains unchanged.

In general, if we're removing any separately compiled functions, we need to be sure that no older user-compiled objects could be looking for those functions (when static linking, or using the import lib).

Reply:
I'm seeing this as test failure. Will fix